thirdparty/quicknode.go

```diff
@@ -60,17 +61,21 @@ func (v *QuicknodeVendor) GetVendorSpecificErrorIfAny(resp *http.Response, jrr i
 				common.NewErrJsonRpcExceptionInternal(code, common.JsonRpcErrorCapacityExceeded, msg, nil, details),
 			)
 		} else if strings.Contains(msg, "failed to parse") {
+			// We do not retry on parse errors, as retrying another upstream would not help.
 			return common.NewErrEndpointClientSideException(
 				common.NewErrJsonRpcExceptionInternal(code, common.JsonRpcErrorParseException, msg, nil, details),
-			)
-		} else if code == -32010 {
+			).WithRetryableTowardNetwork(false)
+		} else if code == -32010 { // Transaction cost exceeds current gas limit
+			// retrying on gas limit exceeded errors toward other upstreams would be helpful, as max gas limit
+			// can be defined per client (reth, geth, parity, etc.) (still needs to be lower than overall block gas limit)
 			return common.NewErrEndpointClientSideException(
 				common.NewErrJsonRpcExceptionInternal(code, common.JsonRpcErrorClientSideException, msg, nil, details),
 			)
-		} else if code == -32602 {
+		} else if code == -32602 && strings.Contains(msg, "cannot unmarshal hex string") {
+			// we do not retry on invalid argument errors, as retrying another upstream would not help.
```

Comment:
We probably should separate these two, an 'invalid argument' could be because Quicknode does not support certain param/tag... which we should retry on other nodes.

The 'cannot unmarshal hex string' is most probably safe to not retry.

Reply:
Am I correct that the `invalid argument (32602)` is a bit too general to be considered retryable?

https://github.com/erpc/erpc/pull/203/files#diff-0b005aa95e710cc341ee8cddc3fb690c956286ea19b90299a23e1c416470fceeR366

Reply:
And also the logic here `else if code == -32602 && strings.Contains(msg, "cannot unmarshal hex string")`

is specific enough to be checked here against quicknode (usage of `&`) and if it is just a `32602 `, it will be checked anyway in `error_normalizer`

Reply:
I think the code is too generic to not retry (i.e. it's better we retry such generic code) but specific texts about param required/unmarshal errors probably is safe to not retry.

In general when in doubt let's retry.